src/libAtomVM/externalterm.c

```diff
@@ -217,6 +219,33 @@ static int serialize_term(Context *ctx, uint8_t *buf, term t)
         }
         return 5;
 
+#ifndef AVM_NO_FP
+    } else if (term_is_float(t)) {
+        if (!IS_NULL_PTR(buf)) {
+            avm_float_t val = term_to_float(t);
+            // ensure interoperable encoding of -0.0
+            if (val == -0.0) {
+                val = 0.0;
+            }
```

Comment:
I am confused about this.

On BEAM:

```
> binary_to_term(<<131,99,"-0.00000000000000000000000000000">>).
-0.0
```

Reply:
I believe there were some platforms in which -0.0 was getting encoded with the high-order sign bit being set, which caused the encoding to be incorrect.  It would be interesting to see what the OTP implementation does.

Reply:
I am still confused with the test. Isn't it run with BEAM?

On BEAM/OTP 25 here:
```
Eshell V13.0  (abort with ^G)
1> erlang:term_to_binary(-0.0).
<<131,70,128,0,0,0,0,0,0,0>>
2> erlang:term_to_binary(0.0). 
<<131,70,0,0,0,0,0,0,0,0>>
```

Reply:
This is somehow reproducing bug OTP-17077.
https://www.erlang.org/patches/otp-24.0#OTP-17077

We need to run our tests on more recent versions of BEAM.

Reply:
Removed this check.  Also, note that the MacOS build does run OTP-25, so at least we have some coverage.

I did find a discrepancy in the OTP 22 vs OTP-25 compilers (maybe OTP-24, as well).  Specifically, the literal `-0.0` gets written by the compiler as `0.0`.  I fixed this in the test by computing `-0.0` in a function, which avoids the compiler issue on all platforms.